src/Engine/ProtoAssociative/CodeGen.cs

```diff
@@ -385,9 +385,12 @@ private bool CyclicDependencyTest(ProtoCore.AssociativeGraph.GraphNode node, ref
 
                 string symbol1 = firstNode.updateNodeRefList[0].nodeList[0].symbol.name;
                 string symbol2 = lastNode.updateNodeRefList[0].nodeList[0].symbol.name;
-                string message = String.Format(ProtoCore.Properties.Resources.kInvalidStaticCyclicDependency, symbol1, symbol2);
 
-                core.BuildStatus.LogWarning(ProtoCore.BuildData.WarningID.kInvalidStaticCyclicDependency, message, core.CurrentDSFileName);
+                foreach (var n in dependencyList.GroupBy(x => x.guid).Select(y => y.First()))
```

Comment:
Exactly. I do not see a proper algorithm to get one node from all the nodes that cause the cyclic dependency